ehcache-impl/src/main/java/org/ehcache/impl/internal/store/heap/SimpleBackend.java

```diff
@@ -105,6 +105,7 @@ public OnHeapValueHolder<V> compute(final K key, final BiFunction<K, OnHeapValue
   public void clear() {
     // This is faster than performing a clear on the underlying map
     realMap = realMapSupplier.get();
+    byteSize.set(0);
```

Comment:
This is going to be problematic when there are other threads modifying the cache at the same time. If wholesale replacement of the map is our strategy we likely need to scope the byteSize counter with the map itself so that they cannot drift apart.

Reply:
I don't think there will be any issue in concurrent execution as well.
`byteSize.set(0)` it's a write operation, even the dirty writes will set byteSize to 0 only.

Reply:
The point here is to make clearing the map and resetting counter as an atomic operation, then other operations modifying cache cannot leave the map and counter in inconsistent state.

Reply:
After a discussion With @AbfrmBlr, got the use case when this could be problematic in concurrent execution.
we need to execute below two statements as atomic :
```
    realMap = realMapSupplier.get();
    byteSize.set(0);
```
One Fix that I applied here, wrapping both the statement under a supplier